DDD-4.md

```diff
@@ -0,0 +1,348 @@
+# DDD-4: Offset Read/Write API
```

Comment:
Do I understand it correctly that we are talking about **REST** API? I suppose from the context but it should be highlighted in the naming as well.

Reply:
There is a REST API aspect to this, which is yet to be documented, but I also was going through each connector and looking at the offsets and what attributes each possessed.  

We need to have a way to pass offset fields with their name, type, and value to the front end, be able to define any relationship between fields or at the very least validate UI input against such relationships, and then have a way where we read/write the offset data to the offset backing store.

As a part of this effort, I feel like we should have some implementation detail in each connector that we can leverage as a part of this REST API, allowing the bulk of the logic for each connector's validation or marshaling to be a detail of the connector itself, not the API needed for the REST extension and/or UI backend.

As for the name, we can certainly find a better naming, I'm open to suggestions.